sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/utils/ResponseBodyModeGeneration.java

```diff
@@ -101,13 +103,47 @@ public static void handleResponseBody(BlockStmt body, TypeMirror returnType, jav
                             + typeElement.getSimpleName() + ".class, " + genericType + ".class);");
                     }
                 }
+            } else {
+                body.addStatement(
+                    "ParameterizedType returnType = CoreUtils.createParameterizedType(" + returnType + ".class);");
+            }
+            boolean isContentTypeSetInHeaders
+                = method.getParameters().stream().anyMatch(parameter -> parameter.getName().equals("contentType"));
+            body.addStatement("Object result = null;");
+            if (isContentTypeSetInHeaders) {
```

Comment:
From prior experience, I wouldn't recommend using the response `Content-Type` here. Instead rely on information given by the annotations and the response type.

For example, Storage allows configuring `Content-Type` of the blob which is returned during download. This code right now runs the risk of failure as we could try to deserialize the blob when we shouldn't.

Reply:
yep, extracting the serialization format from headers and using that instead of this.